tests/test_calc_barycentric_corr.py

```diff
@@ -0,0 +1,247 @@
+        # from time import sleep
+        # sleep(300)
+        im_filepath = "../data/demo"
+        # HACK: workaround for vscode not honoring the cwd specification.
+        from os.path import isdir
+
+        if not isdir(im_filepath):
+            im_filepath = "data/demo"
+        # The demo data has an object moving at x_v=10 px/day
+        # and y_v = 0 px/day. So we search velocities [0, 20]
+        # and angles [-0.5, 0.5].
+        v_arr = [0, 20, 21]
+        ang_arr = [0.5, 0.5, 11]
+
+        self.input_parameters = {
+            # Required
+            "im_filepath": im_filepath,
+            "res_filepath": None,
+            "time_file": None,
+            "output_suffix": "DEMO",
+            "v_arr": v_arr,
+            "ang_arr": ang_arr,
+            "bary_dist": 50.0,
+        }
+
+    def _expected_string(self, baryCoeff, shape_coeff):
+        """Return a string of the actual results in a form that can be copied into baryExpected in the test code. Used for as-built testing."""
+        value_string = f"# fmt: off\nbaryExpected = np.array(["
+        value_string += f"\n    #       dx,                      dxdx,                    dxdy,                    dy,                      dydx,                    dydy,"
+        for i in range(0, shape_coeff[0]):
+            value_string += f"\n    [{baryCoeff[i,0]:23.16f}"
+            for j in range(1, shape_coeff[1]):
+                value_string += f", {baryCoeff[i,j]:23.16f}"
+            value_string += f"],"
+        value_string += f"\n]\n# fmt: on"
+        return value_string
+
+    def _exception_strings(self, baryCoeff, baryExpected, shape_coeff):
+        """Return a list of strings of the actual results that violate the expected results and a boolean that is True if all the results are as expected."""
+        exception_strings = []
+        check_ok = True
+        for i in range(0, shape_coeff[0]):
+            for j in range(1, shape_coeff[1]):
+                if not np.isclose(baryCoeff[i, j], baryExpected[i, j], rtol=1e-5, atol=1e-14):
+                    exception_strings.append(
+                        f"baryCoeff[{i},{j}] = {baryCoeff[i,j]:.16f} != {baryExpected[i,j]:.16f}"
+                    )
+                    check_ok = False
+        return exception_strings, check_ok
+
+    def _check_barycentric(self, baryCoeff, baryExpected):
+        """Check the barycentric correction values against the expected values."""
+        # Check the barycentric correction values
+        shape_coeff = baryCoeff.shape
+        value_string = self._expected_string(baryCoeff, shape_coeff)
+        # print(value_string)
+        shape_expected = baryExpected.shape
+        assert shape_coeff == shape_expected
+        assert len(shape_coeff) == 2
+        exception_strings, check_ok = self._exception_strings(baryCoeff, baryExpected, shape_coeff)
+        if len(exception_strings) > 0:
+            print(f"Exception strings = {exception_strings}")
```

Comment:
Could could return `False` here and then return `True` if the if did not trigger.

Reply:
will change